platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationDirectoryEntryImpl.java

```diff
@@ -0,0 +1,195 @@
+    super(
+        context, JsonUtils.getStringFrom(metadata, MigrationEntryImpl.METADATA_NAME, true), false);
+    this.filtered = JsonUtils.getBooleanFrom(metadata, MigrationEntryImpl.METADATA_FILTERED, false);
+    this.fileEntries =
+        JsonUtils.getListFrom(metadata, MigrationEntryImpl.METADATA_FILES)
+            .stream()
+            .map(Object::toString)
+            .map(FilenameUtils::separatorsToSystem)
+            .map(Paths::get)
+            .map(context::getEntry)
+            .collect(Collectors.toSet());
+    final Long modified =
+        JsonUtils.getLongFrom(metadata, MigrationEntryImpl.METADATA_LAST_MODIFIED, false);
+
+    this.lastModified = (modified != null) ? modified : -1;
+  }
+
+  @Override
+  public long getLastModifiedTime() {
+    return lastModified;
+  }
+
+  @Override
+  public boolean restore(boolean required) {
+    if (restored == null) {
+      super.restored = false; // until proven otherwise
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("Importing {}...", toDebugString());
+      }
+      // a directory is always exported by the framework, as such we can safely extend our
+      // privileges
+      AccessUtils.doPrivileged(
+          () -> {
+            final PathUtils pathUtils = getContext().getPathUtils();
+            final Path apath = getAbsolutePath();
+            // find all existing files ans keep track of it relative from ddf.home to absolute path
+            final Map<Path, Path> existingFiles =
+                FileUtils.listFiles(
+                        apath.toFile(), TrueFileFilter.INSTANCE, TrueFileFilter.INSTANCE)
+                    .stream()
+                    .map(File::toPath)
+                    .collect(Collectors.toMap(pathUtils::relativizeFromDDFHome, p -> p));
+
+            // it is safe to ignore the 'required' parameter since if we get here, we have a
+            // directory
+            // exported to start with and all files underneath are always optional so pass false to
+            // restore()
+            if (fileEntries
+                .stream()
+                .peek(me -> existingFiles.remove(me.getPath()))
+                .map(me -> me.restore(false))
+                .reduce(true, Boolean::logicalAnd)) {
+              if (!filtered) {
+                // all files from the original system were exported under this directory, so remove
+                // all files that were not on the original system but are found on the current one
+                final MigrationReport report = getReport();
+
+                existingFiles.forEach((p, ap) -> PathUtils.cleanQuietly(ap, report));
+                // cleanup all empty directories left underneath this entry's path
+                PathUtils.cleanQuietly(apath, report);
+              }
+              SecurityLogger.audit("Imported directory {}", apath);
+              super.restored = true;
+            } else {
```

Comment:
:question: Do we need to audit on error?

Reply:
We do based on previous comments for the original feature.